kyo-core/shared/src/test/scala/kyo/scheduler/IOPromiseTest.scala

```diff
@@ -291,4 +291,400 @@ class IOPromiseTest extends Test:
         }
     }
 
+    "mask" - {
+        "doesn't propagate interrupts to parent" in {
+            val original = new IOPromise[Nothing, Int]()
+            val masked   = original.mask
+
+            var originalCompleted                         = false
+            var maskedResult: Maybe[Result[Nothing, Int]] = Maybe.Empty
+
+            original.onComplete(_ => originalCompleted = true)
+            masked.onComplete(r => maskedResult = Maybe(r))
+
+            assert(masked.interrupt(Result.Panic(new Exception("Interrupted"))))
+            assert(maskedResult.exists(_.isPanic))
```

Comment:
I think this may be surprising to people coming from ZIO. I generally think the mask should probably cover the masked fiber as well...

Reply:
I was working on changing it but it's not as simple as I thought. Even if we had `onComplete`/`onResult`/`onInterrupt` as discussed privately, we'd still need to have a special `IOPromise` state indicating that it's not interruptible. I'm not sure the complexity and performance cost is worth it. The current implementation follows the behavior of the Twitter Future so it might be ok.

Reply:
Sounds good. We should proceed with this and we can potentially reassess after user feedback.